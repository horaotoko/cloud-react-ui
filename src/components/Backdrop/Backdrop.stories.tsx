import { useState } from "react";
import { Backdrop as BackdropComponent } from "./Backdrop";
import { Button } from "../Button";

export default {
  component: BackdropComponent,
  title: "components/Backdrop",
};

export const Backdrop = (): JSX.Element => {
  const [open, setOpen] = useState(true);

  return (
    <>
      <BackdropComponent open={open} onClick={() => setOpen(false)} />
      <Button onClick={() => setOpen(!open)}>Show Backdrop</Button>
    </>
  );
};
