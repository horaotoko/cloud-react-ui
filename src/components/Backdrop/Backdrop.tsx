import { forwardRef } from 'react';
import styled, { css } from 'styled-components';
import { Backdrop as MuiBackdrop } from '@material-ui/core';
import { CircularProgress } from '../CircularProgress';
import { defaultProps } from '../../theme';

export type Props = {
  open?: boolean;
  onClick?: React.MouseEventHandler<HTMLDivElement>;
  className?: string;
};

const StyledBackdrop = styled(MuiBackdrop)`
  ${({ theme: { backdrop } }) => css`
    && {
      z-index: ${backdrop.zIndex};
    }
  `}
`;
StyledBackdrop.defaultProps = defaultProps;

const Backdrop = forwardRef<HTMLDivElement, Props>(
  ({ open = false, onClick, ...rest }, ref) => (
    <StyledBackdrop ref={ref} open={open} onClick={onClick} {...rest}>
      <CircularProgress />
    </StyledBackdrop>
  )
);

Backdrop.displayName = 'Backdrop';

export { Backdrop };
