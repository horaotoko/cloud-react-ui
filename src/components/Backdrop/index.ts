export { Backdrop } from "./Backdrop";
export type { Props as BackdropProps } from "./Backdrop";
