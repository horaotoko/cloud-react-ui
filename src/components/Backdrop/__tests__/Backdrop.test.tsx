import { render, fireEvent } from '@testing-library/react';
import { Backdrop } from '../Backdrop';

describe('Backdrop', () => {
  it('snapshot test', () => {
    const { asFragment } = render(<Backdrop open onClick={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('called onClick', () => {
    const onClick = jest.fn();
    const { container } = render(<Backdrop open onClick={onClick} />);
    fireEvent.click(container.getElementsByClassName('MuiBackdrop-root')[0]);
    expect(onClick).toBeCalledTimes(1);
  });
});
