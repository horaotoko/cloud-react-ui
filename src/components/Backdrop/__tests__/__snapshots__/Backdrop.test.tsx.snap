// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop snapshot test 1`] = `
<DocumentFragment>
  .c1 .MuiCircularProgress-circle {
  color: #3B7DE9;
}

.c0.c0 {
  z-index: 10;
}

<div
    aria-hidden="true"
    class="MuiBackdrop-root c0"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    theme="[object Object]"
  >
    <div
      class="MuiCircularProgress-root c1 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
      theme="[object Object]"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
